aws/app/lambda/vault_data_integrity_check/vault_data_integrity_check.js

```diff
@@ -0,0 +1,145 @@
+const crypto = require('crypto');
+
+exports.handler = async(event) => {
+
+  try {
+    event.Records
+      .filter(r => r.eventName === "INSERT" || r.eventName === "MODIFY")
```

Comment:
Instead of filtering in the Lambda, for event types and keys, a pattern can be added to the event source mapping in lambda.tf.  This way the lambda is only invoked for messages that you want to check.
https://docs.aws.amazon.com/lambda/latest/dg/invocation-eventfiltering.html